tests/core/providers/test_auto_provider.py

```diff
@@ -123,3 +113,11 @@ def test_web3_auto_infura_websocket_default(monkeypatch, caplog, environ_name):
     w3 = infura.w3
     assert isinstance(w3.provider, WebsocketProvider)
     assert getattr(w3.provider, 'endpoint_uri') == expected_url
+
+
+def test_web3_auto_infura_raises_error_with_nonexistent_scheme(monkeypatch):
+    monkeypatch.setenv('WEB3_INFURA_API_KEY', 'test')
+    monkeypatch.setenv('WEB3_INFURA_SCHEME', 'not-a-scheme')
+
+    with pytest.raises(ValidationError):
```

Comment:
maybe add a `match` keyword to this to ensure the validation error is the expected one (only really relevant if there are multiple paths for `ValidationError` to be raised.